force-app/main/default/classes/OpportunityTriggerHelper.cls

```diff
@@ -0,0 +1,217 @@
+                    if(!plannedSalesToUpdate.isEmpty()){
+                        update plannedSalesToUpdate.values();
+                    }
+                }
+                
+                
+            }
+            
+            
+        }catch (System.NullPointerException e){
+            system.debug(e);
+            System.debug('Error message: ' + e.getMessage()); 
+            System.debug('Stack trace: ' + e.getStackTraceString());
+        }
+    }
+    
+    public static void deleteOpportunity(){
+        
+        List<Opportunity> deletedOpportunities=[SELECT Id,CloseDate,StageName,Amount,(SELECT Id,Product2.Condition__c FROM OpportunityLineItems) FROM Opportunity 
+                                                WHERE StageName='Closed Won' AND Id IN :Trigger.old];
+        
+        Map<Id,Planned_Sales__c> plannedSalesToUpdate=new Map<Id,Planned_Sales__c>();
+        Map<String,Planned_Sales__c> relatedPlannedSales=new Map<String,Planned_Sales__c>();
+        Map<Integer,Integer> monthAndYear=new Map<Integer,Integer>();
+        
+        if(!deletedOpportunities.isEmpty()){
+            for(Opportunity deletedOpportunity:deletedOpportunities){
+                monthAndYear.put(deletedOpportunity.CloseDate.month(),deletedOpportunity.CloseDate.year());
+                
+            }
+            
+            if(!monthAndYear.isEmpty()){
+                for(Planned_Sales__c plannedSale:[SELECT Id,Year__c,Month__c,Car_Type__c,Total_Amount_Earned__c,Count_of_Cars_Sold__c 
+                                                  FROM Planned_Sales__c 
+                                                  WHERE Year__c IN:monthAndYear.values()
+                                                  AND Month__c IN:monthAndYear.keySet()]){
+                                                      
+                                                      String key=plannedSale.Month__c+'_'+plannedSale.Year__c+'_'+plannedSale.Car_Type__c;                             
+                                                      relatedPlannedSales.put(key,plannedSale);
+                                                      
+                                                  }
+                
+                if(!relatedPlannedSales.isEmpty()){
+                    for(Opportunity deletedOpportunity:deletedOpportunities){
+                        String key=deletedOpportunity.CloseDate.month()+'_'+deletedOpportunity.CloseDate.year();
+                        for(OpportunityLineItem oli:deletedOpportunity.OpportunityLineItems){
+                            String key2=key+'_'+oli.Product2.Condition__c;
+                            if(relatedPlannedSales.containsKey(key2)){
+                                Planned_Sales__c plannedSale=relatedPlannedSales.get(key2);
+                                if(plannedSale!=null){
+                                    plannedSale.Count_of_Cars_Sold__c-=deletedOpportunity.OpportunityLineItems.size();
```

Comment:
Adding Opportunity line item amount on each LineItem iteration will not calculate if correctly

If you have 10 cars each 100k -> Opp.Amount = 1 mil, 10 iterations will add 10 mil instead of 1 min

Another thing with Count_of_Cars_Sold__c. You have to only add cars with same Condition as Car Type on planned sales